lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -9639,6 +9639,115 @@ namespace Js
         return superCtor;
     }
 
+    Var JavascriptOperators::OP_ImportCall(__in JavascriptFunction *function, __in Var specifier, __in ScriptContext* scriptContext)
+    {
+        ModuleRecordBase *moduleRecordBase = nullptr;
+        SourceTextModuleRecord *moduleRecord = nullptr;
+
+        FunctionBody* parentFuncBody = function->GetFunctionBody();
+        JavascriptString *specifierString = nullptr;
+
+        try
+        {
+            specifierString = JavascriptConversion::ToString(specifier, scriptContext);
+        }
+        catch (const JavascriptException &err)
+        {
+            Var errorObject = err.GetAndClear()->GetThrownObject(scriptContext);
+            AssertMsg(errorObject != nullptr && Js::JavascriptError::Is(errorObject), "OP_ImportCall: Unhandled exception thrown by ToString(specifier)");
+            if (errorObject != nullptr && Js::JavascriptError::Is(errorObject))
+            {
+                return SourceTextModuleRecord::ResolveOrRejectDynamicImportPromise(false, errorObject, scriptContext);
+            }
+
+            Throw::InternalError();
```

Comment:
I think the reasonable thing to expect is to reject the promise with 1.

Here spec says:
IfAbruptRejectPromise(specifierString, promiseCapability).

which is questionable because specifierString will be nullptr.
I will seek clarification.


---
In reply to: [115379478](https://github.com/Microsoft/ChakraCore/pull/2913#discussion_r115379478) [](ancestors = 115379478)